npe2/_plugin_manager.py

```diff
@@ -187,105 +305,132 @@ def activate(self, key: PluginName) -> PluginContext:
                 if cmd.python_name and cmd.id not in self.commands:
                     self.commands.register(cmd.id, cmd.python_name)
 
+        ctx._activated = True
+        self.events.activation_changed({mf.name}, {})
         return ctx
 
-    def deactivate(self, key: PluginName) -> None:
-        if key not in self._contexts:
+    def get_context(self, plugin_name: PluginName) -> PluginContext:
+        """Return PluginContext for plugin_name"""
+        if plugin_name not in self._contexts:
+            self._contexts[plugin_name] = PluginContext(plugin_name, reg=self.commands)
+        return self._contexts[plugin_name]
+
+    def deactivate(self, plugin_name: PluginName) -> None:
+        """Call the plugin's `on_deactivate` function."""
+        if plugin_name not in self._contexts:
             return
-        mf = self._manifests[key]
-        ctx = self._contexts.pop(key)
+        mf = self._manifests[plugin_name]
+        ctx = self._contexts.pop(plugin_name)
         if mf.on_deactivate:
             _call_python_name(mf.on_deactivate, args=(ctx,))
-            ctx._activated = False
+        ctx._activated = False
         ctx._dispose()
+        self.events.activation_changed({}, {mf.name})
 
-    def iter_compatible_readers(
-        self, path: Union[PathLike, List[PathLike]]
-    ) -> Iterator[ReaderContribution]:
-        if not path:
-            return
+    def enable(self, plugin_name: PluginName) -> None:
+        """Enable a plugin (which mostly means just `un-disable` it.
 
-        if isinstance(path, list):
-            if len({Path(i).suffix for i in path}) > 1:
-                raise ValueError(
-                    "All paths in the stack list must have the same extension."
-                )
-            path = path[0]
+        This is a no-op if the plugin wasn't already disabled.
+        """
+        if not self.is_disabled(plugin_name):
+            return  # pragma: no cover
+
+        self._disabled_plugins.remove(plugin_name)
+        mf = self._manifests.get(plugin_name)
+        if mf is not None:
+            self._contrib.index_contributions(mf)
+        self.events.enablement_changed({plugin_name}, {})
+
+    def disable(self, plugin_name: PluginName) -> None:
+        """Disable a plugin"""
+        self._disabled_plugins.add(plugin_name)
+        self._contrib.remove_contributions(plugin_name)
+        self.events.enablement_changed({}, {plugin_name})
+
+    def is_disabled(self, plugin_name: str) -> bool:
+        """Return `True` if plugin_name is disabled."""
+        return plugin_name in self._disabled_plugins
```

Comment:
I considered that, but that leads to the confusing possibility of `is_enabled('nonsense') == True`  and more often then not, in this case, we're checking to see `is_disabled` rather than `not is_enabled()`

Reply:
good point

Reply:
Ha, I was coming to say this. 
I mean we still can have `is_disabled('nonsens') == False`. So maybe we should error or check that the `plugin_name` is a known plugin ?

Reply:
That's `__contains__`

Reply:
No, I think that's fine.

Reply:
wait, fine to leave it?  or fine to have enabled return true for uninstalled plugins? :joy:?